### PR TITLE
docs: document new process for customizing local ui and custom links

### PR DESCRIPTION
**Pull request by lennessyy**
## Describe the Change

This PR documents the new folder structure required for local UI and custom links.

## Changed Pages

💻 [Add Custom Links to Sidebar](https://deploy-preview-3205--docs-spectrocloud.netlify.app/clusters/edge/local-ui/host-management/custom-link/)

## Jira Tickets

🎫 [PE-4631]()

## Backports

Can this PR be backported?

- [ ] No. This is a release PR.

[PE-4631]: https://spectrocloud.atlassian.net/browse/PE-4631?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docs/docs-content/clusters/edge/local-ui/host-management/custom-link.md
Comment (lennessyy) [inline docs/docs-content/clusters/edge/local-ui/host-management/custom-link.md:151]:
```suggestion
    log in to Local UI.
```

**Comment by lennessyy** [comment]
Removing the breaking change i put in the release notes because i already see it recorded in the release notes PR

File: docs/docs-content/clusters/edge/local-ui/host-management/custom-link.md
Comment (yuliiiah) [inline docs/docs-content/clusters/edge/local-ui/host-management/custom-link.md:67]:
```suggestion
4. Create a directory called **local-ui** at the path **CanvOS/local-ui**. Then, under the **local-ui** directory, create
```

File: docs/docs-content/clusters/edge/local-ui/host-management/custom-link.md
Comment (abhinavnagaraj) [inline docs/docs-content/clusters/edge/local-ui/host-management/custom-link.md:70]: create a file ?

File: docs/docs-content/clusters/edge/local-ui/host-management/custom-link.md
Comment (yuliiiah) [reply docs/docs-content/clusters/edge/local-ui/host-management/custom-link.md:67]: For this step, I would maybe rephrase it a bit differently and give a mkdir command? As in:

> Use the following command to create the **local-ui/ui** directory:
> ```
> mkdir CanvOS/local-ui/ui
> ```

Feels a bit more straightforward then splitting it up into multiple dirs 🤔

File: docs/docs-content/clusters/edge/local-ui/host-management/custom-link.md
Comment (yuliiiah) [inline docs/docs-content/clusters/edge/local-ui/host-management/custom-link.md:72]:
```suggestion
5. In the **CanvOS/local-ui/ui** directory, create a file named **customizations.json**. This is the same file where
   you customize the Local UI theme. If the file is already present, there is no need to create a new one. For more
   information about customizing the Local UI theme, refer to [Customize Local UI Theme](./theming.md).
```

File: docs/docs-content/clusters/edge/local-ui/host-management/custom-link.md
Comment (abhinavnagaraj) [inline docs/docs-content/clusters/edge/local-ui/host-management/custom-link.md:97]: from customizations.json this should be ../app/index.html ?

Comment (yuliiiah) [inline docs/docs-content/clusters/edge/local-ui/host-management/custom-link.md:97]:
```suggestion
   When using a URL to locate the local webpage, you can treat the **local-ui** folder as the root folder and you must
   use absolute paths to locate your assets. For example, from **customizations.json**, you can locate the HTML file in
   the example with `/app/index.html`.
```

File: docs/docs-content/clusters/edge/local-ui/host-management/custom-link.md
Comment (yuliiiah) [inline docs/docs-content/clusters/edge/local-ui/host-management/custom-link.md:145]: Maybe would be good to give a tar command example here? 🤔

File: docs/docs-content/clusters/edge/local-ui/host-management/custom-link.md
Comment (yuliiiah) [inline docs/docs-content/clusters/edge/local-ui/host-management/custom-link.md:150]: > Follow the [Build Edge Artifacts](../../edgeforge-workflow/palette-canvos/palette-canvos.md) guide from step onward

From which step onward? 😅

File: docs/docs-content/clusters/edge/local-ui/host-management/theming.md
Comment (yuliiiah) [inline docs/docs-content/clusters/edge/local-ui/host-management/theming.md:65]: Same thoughts here as above ⬆️

File: docs/docs-content/clusters/edge/local-ui/host-management/theming.md
Comment (yuliiiah) [inline docs/docs-content/clusters/edge/local-ui/host-management/theming.md:89]:
```suggestion
   The `colors.sidebar` property controls the color of the sidebar. This is also the color of the background in the Local UI
   login screen. We suggest you choose a color that contrasts well against your logo as the logo as the color will serve
   as the background for your logo. The **brand** color controls the color of buttons and checkboxes in the UI. The
```

Comment (yuliiiah) [reply docs/docs-content/clusters/edge/local-ui/host-management/theming.md:89]: > We suggest you choose a color that contrasts well against your logo as the logo as the color will serve as the background for your logo.
   
I think something was lost here 🤔 Maybe, _We suggest you choose a color that contrasts well against your logo, because this color will also serve as the logo background._

Comment (yuliiiah) [reply docs/docs-content/clusters/edge/local-ui/host-management/theming.md:89]: Also, we could maybe suggest a tool like [WebAIM](https://webaim.org/resources/contrastchecker/) to check the contrast and recommend aiming for AA at least?

**Review by yuliiiah — APPROVED** [review]
Approved with some edits and suggestions ✨

File: docs/docs-content/clusters/edge/local-ui/host-management/custom-link.md
Comment (lennessyy) [reply docs/docs-content/clusters/edge/local-ui/host-management/custom-link.md:67]: Does this work? I don't think you can create a subdirectory when its parent doesn't exist? But maybe there is a way to override..

```
mkdir CanvOS/local-ui/ui
mkdir: CanvOS/local-ui: No such file or directory
```

File: docs/docs-content/clusters/edge/local-ui/host-management/custom-link.md
Comment (lennessyy) [reply docs/docs-content/clusters/edge/local-ui/host-management/custom-link.md:97]: So I know that you tested that relative links work, but when I used relative links last time, @codrin-iftimie specifically told me that i had to use absolute links. So I used absolute links here. 

@codrin-iftimie Can you confirm that either absolute or relative link will work? Can you tell us why you didn't want me using relative links before?

Comment (abhinavnagaraj) [reply docs/docs-content/clusters/edge/local-ui/host-management/custom-link.md:97]: Since "app" was under "ui" - ui/app/index.html in 4.4, should this be highlighted as a breaking change for existing 4.4 customers using custom-links ?
Someway to indicate that the custom apps are now expected to be at the same level as the "ui" folder instead of being a subfolder, under ui/, when the 4.4 env is upgraded to 4.4a.

Comment (lennessyy) [reply docs/docs-content/clusters/edge/local-ui/host-management/custom-link.md:97]: Yes, this is marked as a breaking change. We document it in the release notes PR [https://github.com/spectrocloud/librarium/pull/3320](Release Notes )

Comment (abhinavnagaraj) [reply docs/docs-content/clusters/edge/local-ui/host-management/custom-link.md:97]: If you see the left side(old) structure, customizations.json and app are in the same folder, so /app/index.html would work. Now app has moved one level up, so we need ../app/index.html as the relative path from customization.json.

Comment (lennessyy) [reply docs/docs-content/clusters/edge/local-ui/host-management/custom-link.md:97]: With absolute paths though, isn't **local-ui** the root folder, and therefore it'd be **/app/index.html** if it's absolute path??

Comment (lennessyy) [reply docs/docs-content/clusters/edge/local-ui/host-management/custom-link.md:97]: I am going to change this to relative path for now with your suggestion. I do want to hear from Codrin on why he wanted me to use absolute path before merging the PR. I'll leave him a message and merge this when I hear back.

Comment (codrin-iftimie) [reply docs/docs-content/clusters/edge/local-ui/host-management/custom-link.md:101]: There are two types of references the customer would use:
1. the link of the application which is something provided in `customizations.json` under  `"link"` attribute for each entry. This needs to be absolute as Lenny mentioned. It can't be relative because the user would access this from the menu from different URLs (eg: on cluster details page `<IP>/cluster/details` we'd append the relative URL which would resolve in a 404 `<IP>/cluster/details/app/index.html`)
2. the applications will have resources that they will reference inside the `index.html` file. This is up to the developer that does this. The good practice is to use absolute URLs paths as well, especially if it's a SPA (single page application)

**Pull request merged**
